main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -1041,10 +1068,19 @@ object Simplifier {
   /**
     * Returns the purity (or impurity) of an expression.
     */
-  private def isPure(eff: Type): Purity = {
+  private def effectToPurity(eff: Type): Purity = {
     if (eff == Type.Pure)
       Pure
     else
       Impure
   }
+
+  /**
+   * Merges purities `p1` and `p2`
+   * A merged purity is only pure if both `p1` and `that` are pure, otherwise it is always impure.
+   */
+  def combinePurities(p1: Purity, p2: Purity): Purity = (p1, p2) match {
```

Comment:
Rename to `combine`. 
Introduce `combine`  function that takes 3 purities (and use it).
Introduce other function `combineAll` that takes a list of purities.